packages/core/src/process/transform/convertDeploymentVertexToExecutionVertex.ts

```diff
@@ -97,13 +99,17 @@ async function convertHardhatContractToContractDeploy(
       vertex.value,
       transformContext
     )) as BigNumber,
+    signer,
   };
 }
 
 async function convertArtifactContractToContractDeploy(
   vertex: ArtifactContractDeploymentVertex,
   transformContext: TransformContext
 ): Promise<ContractDeploy> {
+  const signer: ethers.Signer =
+    await transformContext.services.accounts.getSigner(vertex.from);
```

Comment:
What is the thinking between doing the from -> signer conversion in the graph conversion as against doing it in the execution engine?

Reply:
We discussed this out of band. The conversion has to happen and the conversion phase seems like a sensible location.

There are future questions about what should be in the execution graph, because we may be persisting it to disk as part of a deployment "artifact". That will be the point to consider whether it is signers or a pointer to a signer that is in the execution graph.